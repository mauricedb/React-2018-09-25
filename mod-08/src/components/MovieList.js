import React from 'react';
import MovieListItem from './MovieListItem';

const MovieList = ({ movies }) => (
  <div>
    <h2>Movies</h2>
    <ul>
      {movies.map(m => (
        <MovieListItem key={m.id} movie={m} />
      ))}
    </ul>
  </div>
);

export default MovieList;
