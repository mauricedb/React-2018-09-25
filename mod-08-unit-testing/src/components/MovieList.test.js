import React from 'react';
import reactTestRenderer from 'react-test-renderer';

import MovieList from './MovieList';

test('compare the snapshot', () => {
  const tree = reactTestRenderer
    .create(<MovieList movies={[{ id: 1, title: 'Movie 1' }]} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
