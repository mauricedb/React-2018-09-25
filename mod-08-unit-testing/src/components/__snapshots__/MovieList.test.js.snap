// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compare the snapshot 1`] = `
<div>
  <h2>
    The Cool Movies
  </h2>
  <ul>
    <li
      onClick={[Function]}
    >
      Movie 1
    </li>
  </ul>
</div>
`;
